frame/contracts/proc-macro/src/lib.rs

```diff
@@ -606,6 +607,53 @@ fn expand_functions(def: &EnvDef, expand_blocks: bool, host_state: TokenStream2)
 		let is_stable = f.is_stable;
 		let not_deprecated = f.not_deprecated;
 
+		// wrapped host function body call with traces when target is`runtime::contracts::strace` and level is `trace`
+        // e.g ./target/release/substrate-contracts-node -l "fatal,runtime::contracts::strace=trace" --dev
+		//WIP(pg): replace comment above with link once the doc has been added to the readme
+		let wrapped_body_with_trace = {
+			let params_fmt_str = params.clone().filter_map(|arg| match arg {
+				syn::FnArg::Receiver(_) => None,
+				syn::FnArg::Typed(p) => {
+					match *p.pat.clone() {
+						syn::Pat::Ident(ref pat_ident) => Some(pat_ident.ident.to_string()),
+						_ => None,
+					}
+				},
+			})
+			.map(|s| format!("{s}: {{:?}}"))
+			.collect::<Vec<_>>()
+			.join(", ");
+
+			let trace_fmt_str = format!("{}::{}({}) = {{:?}}", module, name, params_fmt_str);
+			let trace_fmt_args = params.clone().filter_map(|arg| match arg {
+				syn::FnArg::Receiver(_) => None,
+				syn::FnArg::Typed(p) => {
+					match *p.pat.clone() {
+						syn::Pat::Ident(ref pat_ident) => Some(pat_ident.ident.clone()),
+						_ => None,
+					}
+				},
+			});
+
+			let debug_buffer_fmt_str = format!("{}\n", trace_fmt_str);
+			let debug_buffer_fmt_args = trace_fmt_args.clone();
+
+			quote! {
+				if ::log::log_enabled!(target: "runtime::contracts::strace", ::log::Level::Trace) {
+					let result = #body;
+					
+					::log::trace!(target: "runtime::contracts::strace", #trace_fmt_str, #( #trace_fmt_args, )* result);
+					#[cfg(feature = "std")]
+					ctx.ext.append_debug_buffer(&format!(#debug_buffer_fmt_str, #( #debug_buffer_fmt_args, )* result));
+					result
```

Comment:
this is not structured, for now
I am thinking there is no point into using json here, until the debug view of contracts-ui can parse it and display it properly. What do you think?

Reply:
The debug buffer is just text for human consumption. So yes. No json. It is not meant to be parsed by machines.